src/main/java/org/opentripplanner/middleware/cdp/ConnectedDataManager.java

```diff
@@ -0,0 +1,216 @@
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+import static org.opentripplanner.middleware.utils.ConfigUtils.getConfigPropertyAsInt;
+import static org.opentripplanner.middleware.utils.ConfigUtils.getConfigPropertyAsText;
+import static org.opentripplanner.middleware.utils.DateTimeUtils.DEFAULT_DATE_FORMAT_PATTERN;
+import static org.opentripplanner.middleware.utils.DateTimeUtils.convertToLocalDate;
+import static org.opentripplanner.middleware.utils.DateTimeUtils.getDateMinusNumberOfDays;
+import static org.opentripplanner.middleware.utils.DateTimeUtils.getDatePlusNumberOfDays;
+import static org.opentripplanner.middleware.utils.DateTimeUtils.getEndOfDay;
+import static org.opentripplanner.middleware.utils.DateTimeUtils.getStartOfDay;
+import static org.opentripplanner.middleware.utils.DateTimeUtils.getStringFromDate;
+
+public class ConnectedDataManager {
+
+    // If set to true, no files are upload to S3 or deleted from the local disk (this will be carried out by the unit
+    // tests instead).
+    public static boolean IS_TEST;
+
+    public static final String FILE_NAME_SUFFIX = "anon-trip-data";
+    public static final String ZIP_FILE_NAME_SUFFIX = FILE_NAME_SUFFIX + ".zip";
+    public static final String DATA_FILE_NAME_SUFFIX = FILE_NAME_SUFFIX + ".json";
+
+    private static final int CONNECTED_DATA_PLATFORM_TRIP_HISTORY_UPLOAD_JOB_FREQUENCY_IN_MINUTES =
+        getConfigPropertyAsInt("CONNECTED_DATA_PLATFORM_TRIP_HISTORY_UPLOAD_JOB_FREQUENCY_IN_MINUTES", 5);
+
+    private static final Logger LOG = LoggerFactory.getLogger(ConnectedDataManager.class);
+
+    public static final String CONNECTED_DATA_PLATFORM_S3_BUCKET_NAME =
+        getConfigPropertyAsText("CONNECTED_DATA_PLATFORM_S3_BUCKET_NAME");
+
+    public static final String CONNECTED_DATA_PLATFORM_S3_FOLDER_NAME =
+        getConfigPropertyAsText("CONNECTED_DATA_PLATFORM_S3_FOLDER_NAME");
+
+    public static void scheduleTripHistoryUploadJob() {
+        LOG.info("Scheduling trip history upload for every {} minute(s)", CONNECTED_DATA_PLATFORM_TRIP_HISTORY_UPLOAD_JOB_FREQUENCY_IN_MINUTES);
+        Scheduler.scheduleJob(
+            new TripHistoryUploadJob(),
+            0,
+            CONNECTED_DATA_PLATFORM_TRIP_HISTORY_UPLOAD_JOB_FREQUENCY_IN_MINUTES,
+            TimeUnit.MINUTES);
+    }
+
+    /**
+     * Remove a user's trip requests and trip summaries from the database. Record the user's trip dates so that all
+     * data previously uploaded to s3 can be recompiled and uploaded again.
+     */
+    public static void removeUsersTripHistory(String userId) {
+        Set<Date> userTripDates = new HashSet<>();
+        for (TripRequest request : TripRequest.requestsForUser(userId)) {
+            userTripDates.add(getStartOfDay(request.dateCreated));
+            // This will delete the trip summaries as well.
+            request.delete();
```

Comment:
Is there a way to do a bulk-delete instead of a bunch of individual deletes?

Reply:
I could go with `removeFiltered(eq("userId", userId))`, but I would still need to loop through each in order to get the trip request id so I could delete the related trip summaries.